### DISCUSSION TITLE
Fabric Object controllers.

### DISCUSSION THREAD

**Discussion by Umairkhan07** [fabricjs/fabric.js#8460] [General]
Hi there, I am trying to increase the detection area of fabric Js object controllers. Is there any way to do this? 
its something like padding. what I am trying to say is suppose object controller corner size is set to 20 and I want to detect this corner when mouse is at radius of 30 or 40 of the actual corner size.

**Comment by vipin-vidyut** [comment]
fabric.Object.prototype.cornerSize = 50; // Or what size you need

**Reply by Umairkhan07** [reply]
fabric.Object.prototype.cornerSize = 50; // will increase the size of the corner, I only want to increases controller detection area not the actual corner size. suppose increasing padding but the actual coloured area remains smaller.

**Reply by Umairkhan07** [reply]
is there anything like fabric.Object.prototype.controls.tl.padding = 30; ?

**Reply by vipin-vidyut** [reply]
http://fabricjs.com/controls-customization

**Reply by Umairkhan07** [reply]
I have already seen that, but its not there what I want. Thank you for sharing.

**Reply by Umairkhan07** [reply]
https://codepen.io/umairu07/pen/eYKMzrG?editors=0011 
this is what I want. can you help me with how to do this in react? 
note : look how mouse detects the controllers from some distance.

**Comment by ShaMan123** [comment]
I am surprised this feature is not supported. I was sure it was...
Looked at the source code and didn't come up with something.
It should be fairly simple adding this feature. Look at object interactivity and geometry mixins. Coords are what matter. But somewhere, probably in canvas_events mixin.__onMouseDown is the logic that chooses the control.
Try submitting a pr. I can help.
